scripts/install/opt/seagate/cortx/motr/libexec/motr_data_recovery.sh

```diff
@@ -882,19 +903,33 @@ is_user_root_user # check the script is running with root access
 
 shutdown_services
 
-run_cmd_on_local_node "./prov-m0-reset $CDF_FILENAME $HA_ARGS_FILENAME --start-consul"
+run_cmd_on_local_node "./prov-m0-reset $CDF_FILENAME $HA_ARGS_FILENAME --start-consul $SINGLE_NODE_RUNNING"
 
 get_cluster_configuration
 
-run_cmd_on_local_node "./prov-m0-reset $CDF_FILENAME $HA_ARGS_FILENAME --stop-consul"
+run_cmd_on_local_node "./prov-m0-reset $CDF_FILENAME $HA_ARGS_FILENAME --stop-consul $SINGLE_NODE_RUNNING"
 
-# Execute becktool only in the script if option --beck-only is set
+# Execute becktool only in the script if option --beck-only is set.
+# Use --beck-only option only when snapshot is already present on the system.
 if $beck_only ; then
     m0drlog "Running only becktool in the recovery script"
     run_becktool
     exit $?
 fi
 
+# Only remove MD_Snapshot if present and create lv_main_swap again
+if $clean_snapshot ; then
+    m0drlog "Removing MD_Snapshot if present and create lv_main_swap again"
+    remove_snapshot                # remove snapshot on both nodes
+    remove_snapshot_status=$?
+    [[ remove_snapshot_status -eq 0 ]] || { die "ERROR: Remove snapshot failed with code $remove_snapshot_status"; }
```

Comment:
This can be written as:
```
remove_snapshot ||  { die "ERROR: Remove snapshot failed with code $r?" }
```

Reply:
the same for "create_swap"

Reply:
I will keep the changes for now but make a note of your recommendation for Kanchan when he comes back on Monday. In the mean time please accept the current change so that the review is not stalled.